packages/build-tools/src/context.ts

```diff
@@ -98,7 +98,18 @@ export class BuildContext<TJob extends Job = Job> {
     this.cacheManager = options.cacheManager;
     this._uploadArtifact = options.uploadArtifact;
     this.reportError = options.reportError;
-    this._job = job;
+
+    const shouldApplyRepackOverrides = job.platform && job.mode === BuildMode.REPACK;
```

Comment:
Why do we need the `job.platform` check?

Reply:
Because `Generic.Job` doesn't support `mode`

Reply:
Ah, sorry, I thought I'm looking at `BuildContext` in `launcher` where `job` is guaranteed to be `BuildJob`.